libp2p/utility.nim

```diff
@@ -115,21 +115,24 @@ template withValue*[T](self: Opt[T] | Option[T], value, body: untyped): untyped
 macro withValue*[T](self: Opt[T] | Option[T], value, body, body2: untyped): untyped =
   let elseBody = body2[0]
   quote do:
-    if `self`.isSome:
-      let `value` {.inject.} = `self`.get()
+    let temp = `self`
```

Comment:
We fixed the template but didn't notice the macro. Should we use `()` as in the template? I don't remember why it's necessary. Could you please rename `body2` to a more descriptive name and add docs similar to the template?

Reply:
The `()` are necessary in C macro when you're doing something like that for example
```C
#include <stdio.h>
#define Mult_by_two(x) x * 2
#define Mult_by_two_parenthesis(x) (x) * 2

int main(void) {
  printf("%i\n", Mult_by_two(2+2)); // will print 6 (2 + 2 * 2);
  printf("%i\n", Mult_by_two_parenthesis(2+2)); // will print 8 ((2+2)*2);
  return (0);
}
```
And I usually tend to use this in the other languages, even if the language fix this "issue".